### PR TITLE
Test/week 7/integration branch

### PR DESCRIPTION
**Pull request by mirapanigrahy**
@AG6GR I made the changes you said for the drive code in this new branch

### CODE REVIEW

File: src/main/java/frc/robot/systems/DriveFSMSystem.java [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+		rearLeft.setDesiredState(swerveModuleStates[2]);
+		rearRight.setDesiredState(swerveModuleStates[(2 + 1)]);
+	}
+
+	/**
+	 * Balances gyro.
+	 */
+	public void balance() {
+		double power;
+		if (Math.abs(gyro.getRoll()) < 2 && Math.abs(gyro.getRoll()) > (-1 * 2)) {
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/DriveFSMSystem.java]: Simplify this expression. Absolute value is always positive.

File: src/main/java/frc/robot/systems/DriveFSMSystem.java [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+			power = 0;
+		} else if (gyro.getRoll() > 0) {
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/DriveFSMSystem.java]: These last three cases are exactly the same. If val < 0, then a + val is the same as a - abs(value).

Also spelling for BALENCE_SPEED_INVERSE_PROPORTION

File: src/main/java/frc/robot/systems/DriveFSMSystem.java [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+	public double getHeading() {
+		return Rotation2d.fromDegrees(-gyro.getAngle()).getDegrees();
+	}
+
+	/**
+	 * Returns the turn rate of the robot.
+	 *
+	 * @return The turn rate of the robot, in degrees per second
+	 */
+	public double getTurnRate() {
```
Comment (AG6GR) [inline src/main/java/frc/robot/systems/DriveFSMSystem.java]: Is this ever used?

**Review by AG6GR — COMMENTED** [review]
For future reference you can push additional commits to a branch and the PR will update automatically, no need to create a new PR.

I've added back the comments that haven't been resolved. Also please add the citations for code provided by REV. I recommend adding a link to the original source repo below the original copyright comment and mentioning it in the README file.

**Comment by mirapanigrahy** [comment]
> For future reference you can push additional commits to a branch and the PR will update automatically, no need to create a new PR.
> 
> I've added back the comments that haven't been resolved. Also please add the citations for code provided by REV. I recommend adding a link to the original source repo below the original copyright comment and mentioning it in the README file.

@AG6GR I made the changes

**Review by AG6GR — APPROVED** [review]

**Pull request merged**